src/backend/cuda/kernel/random_engine.hpp

```diff
@@ -0,0 +1,655 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <dispatch.hpp>
+#include <err_cuda.hpp>
+#include <debug_cuda.hpp>
+#include <af/defines.h>
+#include <kernel/random_engine_philox.hpp>
+#include <kernel/random_engine_threefry.hpp>
+#include <kernel/random_engine_mersenne.hpp>
+#include <random_engine.hpp>
+
+namespace cuda
+{
+namespace kernel
+{
+    //Utils
+
+    static const int THREADS = 256;
+    #define UINTMAXFLOAT 4294967296.0f
+    #define UINTLMAXDOUBLE (4294967296.0*4294967296.0)
+    #define PI_VAL 3.1415926535897932384626433832795028841971693993751058209749445923078164
```

Comment:
Don't we have PI defined already? https://github.com/arrayfire/arrayfire/blob/2ecff9818f23c2b0fd35ab992d4e00f17fa039b9/include/af/constants.h#L15